src/hooks/useWalletSupportsChain/useWalletSupportsChain.ts

```diff
@@ -29,27 +29,41 @@ import {
   supportsPolygon,
   supportsThorchain,
 } from '@shapeshiftoss/hdwallet-core'
+import { isLedger } from '@shapeshiftoss/hdwallet-ledger'
 import { MetaMaskShapeShiftMultiChainHDWallet } from '@shapeshiftoss/hdwallet-shapeshift-multichain'
 import { useMemo } from 'react'
 import { useIsSnapInstalled } from 'hooks/useIsSnapInstalled/useIsSnapInstalled'
+import { selectAccountIdsByChainId } from 'state/slices/portfolioSlice/selectors'
+import { useAppSelector } from 'state/store'
 
 type WalletSupportsChainArgs = {
   isSnapInstalled: boolean | null
+  chainAccountIds: AccountId[] // allows dynamic chain-support detection for Ledger
   chainId: ChainId
   wallet: HDWallet | null
 }
 
 // use outside react
 export const walletSupportsChain = ({
   chainId,
+  chainAccountIds,
   wallet,
   isSnapInstalled,
 }: WalletSupportsChainArgs): boolean | null => {
   if (!wallet) return false
   const isMetaMaskMultichainWallet = wallet instanceof MetaMaskShapeShiftMultiChainHDWallet
   // Naming is slightly weird there, but the intent is if this evaluates to false, it acts as a short circuit
-  const shortCircuitFeatureDetection =
-    !isMetaMaskMultichainWallet || (isMetaMaskMultichainWallet && isSnapInstalled)
+  const shortCircuitFeatureDetection = (() => {
```

Comment:
We might want to invert this logic, and call it something like `shouldShortCircuitFeatureDetection`.

Then in the switch statement below, `&& !shouldShortCircuitFeatureDetection`.

Reply:
As discussed, renamed this to `hasRuntimeSupport` to both convey intent better, and avoid inverting this logic 54d80e56196f5f0dd2ba9f6feae54d1df3f24ca6